File: chap2/js/linearSearch.js
const numbers = [31,41,59,26,41,58];
const value1 = 26;
const value2 = 100;
let find = false;

function linearSearch(numbers, value){
    console.log(!test());

    for(const i in numbers){
        if(numbers[i] === value){
            find = true;
            console.log(test());
            return i;
        }
        console.log(!test());
    }
    console.log(!test());
    return -1;
}

function test(){ return find; }

console.log("Index: " + linearSearch(numbers, value1));
find = false;
console.log("Index: " + linearSearch(numbers, value2));

File: chap2/c++/linear_search.cpp
#include <iostream>

#define SIZE 6
bool find = false;

bool test(){
    return find;
}

int linear_search(int* numbers, int value){
    std::cout << !test() << std::endl;

    for(int i = 0; i < SIZE; i++){
        if(numbers[i] == value){
            find = true;
            std::cout << test() << std::endl;
            return i;
        }
        std::cout << !test() << std::endl;
    }
    std::cout << !test() << std::endl;
    return -1;
}

int main(void){

    int numbers[SIZE] = {31,41,59,26,41,58};
    int value1 = 26;
    int value2 = 100;

    std::cout << "Index: " << linear_search(numbers, value1) << std::endl;
    find = false;
    std::cout << "Index: " << linear_search(numbers, value2) << std::endl;

    return 0;
}